src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -329,10 +346,16 @@ private bool IsString(int typeIndex)
             return typeIndex == _stringTypeIndex;
         }
 
-        private static byte[] EncodeStringName()
+        private static byte[] EncodeStringName(bool fLegacyName)
         {
             var type = typeof(string);
             var name = type.AssemblyQualifiedName;
```

Comment:
@jkotas From ResourceReader/Writer standpoint, we always write to read/write mscorlib for compat purposes? I was not sure and hence I added check for both.